components/style/properties/longhand/font.mako.rs

```diff
@@ -1006,3 +1006,865 @@ ${helpers.single_keyword("font-variant-position",
+                            let args = try!(input.parse_comma_separated(|input| {
+                                Ok(try!(input.expect_ident()).into_owned())
+                            }));
+                            character_variant = Some(args);
+                            Ok(())
+                        }));
+                    },
+                    "swash" => {
+                        if swash.is_some() {
+                            return Err(());
+                        }
+
+                        try!(input.parse_nested_block(|input| {
+                            let arg = try!(input.expect_ident()).into_owned();
+                            swash = Some(arg);
+                            Ok(())
+                        }));
+                    },
+                    "ornaments" => {
+                        if ornaments.is_some() {
+                            return Err(());
+                        }
+
+                        try!(input.parse_nested_block(|input| {
+                            let arg = try!(input.expect_ident()).into_owned();
+                            ornaments = Some(arg);
+                            Ok(())
+                        }));
+                    },
+                    "annotation" => {
+                        if annotation.is_some() {
+                            return Err(());
+                        }
+
+                        try!(input.parse_nested_block(|input| {
+                            let arg = try!(input.expect_ident()).into_owned();
+                            annotation = Some(arg);
+                            Ok(())
+                        }));
+                    },
+                    _ => break, 
+                };
+            } else {
+                break;
+            }
+        }
+
+        Ok(SpecifiedValue::Style(Style{
+            historical_forms: historical_forms,
+            styleset: styleset,
+            character_variant: character_variant,
+            swash: swash,
+            ornaments: ornaments,
+            annotation: annotation,
+        }))
+    }
+
+    #[inline]
+    pub fn get_initial_value() -> computed_value::T {
+        computed_value::T::Normal
+    }
+
+    #[inline]
+    pub fn get_initial_specified_value() -> SpecifiedValue {
+        SpecifiedValue::Normal
+    }
+</%helpers:longhand>
+
+<%helpers:longhand name="font-variant-east-asian" products="none" animatable="False"
+                   spec="https://drafts.csswg.org/css-fonts-3/#font-variant-east-asian-prop">
+    use std::fmt;
+    use style_traits::ToCss;
+    use values::HasViewportPercentage;
+    use values::computed::ComputedValueAsSpecified;
+
+    impl ComputedValueAsSpecified for SpecifiedValue {}
+    no_viewport_percentage!(SpecifiedValue);
+
+    pub mod computed_value {
+        pub use super::SpecifiedValue as T;
+    }
+
+    #[derive(Debug, Clone, PartialEq)]
+    #[cfg_attr(feature = "servo", derive(HeapSizeOf))]
+    pub enum VariantValue {
```

Comment:
This is a keyword enum, right? Why can't we use `define_css_keyword_enum!`? (which comes with `ToCss` and `Parse` impls for free)